### PR TITLE
Azure gov support

### PR DESCRIPTION
**Pull request by bathina2**
The following PR adds support for azure gov cloud by allowing us to set the environment name when configuring the azure government client.

### CODE REVIEW

File: azure/config.go [hunk 1]
```diff
@@ -69,7 +71,20 @@ func newBlobStorageClient(cfg stow.Config) (*az.BlobStorageClient, error) {
...
-	basicClient, err := az.NewBasicClient(acc, key)
+
+	var basicClient az.Client
+	var err error
+	envName, ok := cfg.Config(ConfigEnvName)
+	if ok && envName != "" {
+		env, err := azure.EnvironmentFromName(envName)
+		if err != nil {
+			return nil, err
+		}
```
Comment (DeepikaDixit) [inline azure/config.go:82]: Is this required? The other `err` assignment is in the `else` block so both will eventually go to line 88

Comment (bathina2) [reply azure/config.go:82]: If we don't return this right away it may get  overwritten by the err on 83.

Comment (bathina2) [reply azure/config.go:82]: And that error may not be as descriptive.

Comment (DeepikaDixit) [reply azure/config.go:82]: AhhH 🤦‍♀️  Missed that. I read it as `else` after this line. My bad.

**Review by DeepikaDixit — APPROVED** [review]
LGTM

**Review by onkarbhat — APPROVED** [review]

**Review by miquella — APPROVED** [review]

**Review by leuyentran — APPROVED** [review]

**Pull request merged**